HTTP connector in `camunda/connectors:0.2.0-rc1-ootb` image fails to evaluate response mapping expression

__Describe the Bug__

I'm trying to run the ootb connectors via `camunda/connectors:0.2.0-rc1-ootb`.

Using the rest connector with a result mapping fails:

```
Sep 29, 2022 10:51:16 AM io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler handle
INFO: Received job 2251799813851702
Sep 29, 2022 10:51:16 AM io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler handle
SEVERE: Failed to process job 2251799813851702
io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapperException: Failed to evaluate expression '=result.body.id' and context 'HttpJsonResult{status=200, headers={date=Thu, 29 Sep 2022 10:51:16 GMT, transfer-encoding=chunked, keep-alive=timeout=60, content-type=application/vnd.spring-boot.actuator.v3+json, connection=keep-alive}, body={id=310}}', because expression evaluation failed
	at io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapper.evaluateToJson(FeelEngineWrapper.java:84)
	at io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler.lambda$createOutputVariables$0(ConnectorJobHandler.java:97)
	at java.base/java.util.Optional.map(Unknown Source)
	at io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler.createOutputVariables(ConnectorJobHandler.java:97)
	at io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler.handle(ConnectorJobHandler.java:73)
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.executeJob(JobRunnableFactory.java:44)
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.lambda$create$0(JobRunnableFactory.java:39)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.ClassCastException: class io.camunda.connector.http.model.HttpJsonResult cannot be cast to class java.util.Map (io.camunda.connector.http.model.HttpJsonResult is in unnamed module of loader 'app'; java.util.Map is in module java.base of loader 'bootstrap')
	at io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapper.ensureVariablesMap(FeelEngineWrapper.java:70)
	at io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapper.evaluateToJson(FeelEngineWrapper.java:75)
	... 12 more
```

The expression I used is `= result.body.id`. In the exception we can see that this variable is available.

__Steps to Reproduce__

1. Deploy `camunda/connectors:0.2.0-rc1-ootb`
2. Use the HTTP connector with a response mapping expression


__Expected Behavior__

The HTTP connector in the OOTB docker image can evaulate response mapping expression.

@igpetrov Thanks for looking into it! :+1: 

> Also, the default result context is stored in the response variable

:facepalm: Thanks, that was my mistake. However I'm still getting the same exception after adjusting the expression. Here is a simple model to reproduce:

[simple-rest-process.zip](https://github.com/camunda/connector-sdk/files/9706932/simple-rest-process.zip)

The expression is now just `=response.body`, mapped to a new variable called `body`. AFAIK this should work?

The logs from the connectors container look like there are classloader issues:

```
io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapperException: Failed to evaluate expression '=response.body' and context 'HttpJsonResult{status=200, headers={date=Tue, 04 Oct 2022 12:37:03 GMT, transfer-encoding=chunked, keep-alive=timeout=60, content-type=application/vnd.spring-boot.actuator.v3+json, connection=keep-alive}, body={backupId=110, status=COMPLETED, failureReason=java.lang.Object@28814363, partitions=[{partitionId=1, status=COMPLETED, description={snapshotId=, checkpointPosition=29208, brokerId=1, brokerVersion=8.1.0}, failureReason=java.lang.Object@28814363, createdAt=2022-10-04T12:29:49.824635836Z, lastUpdatedAt=2022-10-04T12:29:50.562648101Z}, {partitionId=2, status=COMPLETED, description={snapshotId=, checkpointPosition=130946, brokerId=1, brokerVersion=8.1.0-SNAPSHOT}, failureReason=java.lang.Object@28814363, createdAt=2022-09-29T10:36:42.079653175Z, lastUpdatedAt=2022-09-29T10:36:44.059903006Z}, {partitionId=3, status=COMPLETED, description={snapshotId=338116-1-1026825-1025098, checkpointPosition=3020625, brokerId=2, brokerVersion=8.1.0-SNAPSHOT}, failureReason=java.lang.Object@28814363, createdAt=2022-09-29T07:19:53.715945204Z, lastUpdatedAt=2022-09-29T07:20:24.985795057Z}]}}', because expression evaluation failed
	at io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapper.evaluateToJson(FeelEngineWrapper.java:84)
	at io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler.lambda$createOutputVariables$0(ConnectorJobHandler.java:97)
	at java.base/java.util.Optional.map(Unknown Source)
	at io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler.createOutputVariables(ConnectorJobHandler.java:97)
	at io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler.handle(ConnectorJobHandler.java:73)
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.executeJob(JobRunnableFactory.java:44)
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.lambda$create$0(JobRunnableFactory.java:39)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.ClassCastException: class io.camunda.connector.http.model.HttpJsonResult cannot be cast to class java.util.Map (io.camunda.connector.http.model.HttpJsonResult is in unnamed module of loader 'app'; java.util.Map is in module java.base of loader 'bootstrap')
	at io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapper.ensureVariablesMap(FeelEngineWrapper.java:70)
	at io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapper.evaluateToJson(FeelEngineWrapper.java:75)
	... 12 more
```

I can reproduce the behavior described from Ole, using the self managed distribution from https://github.com/camunda/camunda-platform/pull/35 and the following bpmn model [http-post.bpmn.txt](https://github.com/camunda/connector-sdk/files/9708051/http-post.bpmn.txt)

The model works as expected in SaaS but fails in Self-Managed with the following exception
```
SEVERE: Failed to process job 2251799813685265
io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapperException: Failed to evaluate expression '={ json: response.body.json }' and context 'HttpJsonResult{status=200, headers={date=Tue, 04 Oct 2022 14:41:06 GMT, access-control-allow-origin=*, content-length=467, server=gunicorn/19.9.0, access-control-allow-credentials=true, content-type=application/json, connection=keep-alive}, body={args={}, data={"hello":"world"}, files={}, form={}, headers={Accept-Encoding=gzip, Content-Length=17, Content-Type=application/json, Host=httpbin.org, User-Agent=Google-HTTP-Java-Client/1.42.2 (gzip), X-Amzn-Trace-Id=Root=1-633c4602-30a9825f4a7f5ec656c2f958}, json={hello=world}, origin=79.202.44.92, url=https://httpbin.org/post}}', because expression evaluation failed
        at io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapper.evaluateToJson(FeelEngineWrapper.java:84)
        at io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler.lambda$createOutputVariables$0(ConnectorJobHandler.java:96)
        at java.base/java.util.Optional.map(Unknown Source)
        at io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler.createOutputVariables(ConnectorJobHandler.java:96)
        at io.camunda.connector.runtime.jobworker.api.outbound.ConnectorJobHandler.handle(ConnectorJobHandler.java:72)
        at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.executeJob(JobRunnableFactory.java:44)
        at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.lambda$create$0(JobRunnableFactory.java:39)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.ClassCastException: class io.camunda.connector.http.model.HttpJsonResult cannot be cast to class java.util.Map (io.camunda.connector.http.model.HttpJsonResult is in unnamed module of loader 'app'; java.util.Map is in module java.base of loader 'bootstrap')
        at io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapper.ensureVariablesMap(FeelEngineWrapper.java:70)
        at io.camunda.connector.runtime.jobworker.impl.feel.FeelEngineWrapper.evaluateToJson(FeelEngineWrapper.java:75)
        ... 12 more
```

It could be a class loading issue, or related with job worker implementation to handle result expressions. I will root cause the issue further.

--- a/runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/feel/FeelEngineWrapper.java
+++ b/runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/feel/FeelEngineWrapper.java
@@ -17,6 +17,7 @@
 package io.camunda.connector.runtime.jobworker.impl.feel;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.module.scala.DefaultScalaModule$;
 import java.util.HashMap;
@@ -31,6 +32,8 @@ public class FeelEngineWrapper {
   static final String RESPONSE_MAP_KEY = "response";
   static final String ERROR_VARIABLES_MUST_NOT_BE_NULL = "variables cannot be null";
   static final String ERROR_EXPRESSION_EVALUATION_FAILED = "expression evaluation failed";
+  public static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE =
+      new TypeReference<>() {};
 
   private final FeelEngine feelEngine;
   private final ObjectMapper objectMapper;
@@ -64,10 +67,9 @@ public class FeelEngineWrapper {
     return scala.collection.immutable.Map.from(CollectionConverters.asScala(context));
   }
 
-  @SuppressWarnings("unchecked")
-  private static Map<String, Object> ensureVariablesMap(final Object variables) {
-    return (Map<String, Object>)
-        Objects.requireNonNull(variables, ERROR_VARIABLES_MUST_NOT_BE_NULL);
+  private Map<String, Object> ensureVariablesMap(final Object variables) {
+    Objects.requireNonNull(variables, ERROR_VARIABLES_MUST_NOT_BE_NULL);
+    return objectMapper.convertValue(variables, MAP_TYPE_REFERENCE);
   }
 
   public String evaluateToJson(final String expression, final Object variables) {

--- a/runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/feel/FeelEngineWrapper.java
+++ b/runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/feel/FeelEngineWrapper.java
@@ -32,8 +32,8 @@ public class FeelEngineWrapper {
   static final String RESPONSE_MAP_KEY = "response";
   static final String ERROR_VARIABLES_MUST_NOT_BE_NULL = "variables cannot be null";
   static final String ERROR_EXPRESSION_EVALUATION_FAILED = "expression evaluation failed";
-  public static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE =
-      new TypeReference<>() {};
+
+  static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE = new TypeReference<>() {};
 
   private final FeelEngine feelEngine;
   private final ObjectMapper objectMapper;

--- a/runtime-job-worker/src/test/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandlerTest.java
+++ b/runtime-job-worker/src/test/java/io/camunda/connector/runtime/jobworker/api/outbound/ConnectorJobHandlerTest.java
@@ -115,6 +115,25 @@ public class ConnectorJobHandlerTest {
     }
 
     @Test
+    public void shouldSetToResultExpressionWhenPojoIsReturned() {
+      // given
+      // Response from service -> {"callStatus":{"statusCode":"200 OK"}}
+      final String responseValue = "response";
+      var jobHandler =
+          new ConnectorJobHandler((context) -> new TestConnectorResponsePojo(responseValue));
+
+      // FEEL expression -> {"processedOutput":response.callStatus}
+      final String resultExpression = "{\"processedOutput\": response.value }";
+
+      // when
+      var result =
+          JobBuilder.create().withResultExpressionHeader(resultExpression).execute(jobHandler);
+
+      // then
+      assertThat(result.getVariables()).isEqualTo(Map.of("processedOutput", responseValue));
+    }
+
+    @Test
     public void shouldSetBothResultVariableAndExpression() {
       // given
       // Response from service -> {"callStatus":{"statusCode":"200 OK"}}
@@ -211,6 +230,18 @@ public class ConnectorJobHandlerTest {
     @Override
     public SecretProvider getSecretProvider() {
       return name -> TestSecretProvider.SECRET_NAME.equals(name) ? "baz" : null;
+    }
+  }
+
+  private static class TestConnectorResponsePojo {
+    private final String value;
+
+    private TestConnectorResponsePojo(final String value) {
+      this.value = value;
+    }
+
+    public String getValue() {
+      return value;
     }
   }
 }

--- a/runtime-job-worker/src/test/java/io/camunda/connector/runtime/jobworker/impl/feel/FeelEngineWrapperExpressionEvaluationTest.java
+++ b/runtime-job-worker/src/test/java/io/camunda/connector/runtime/jobworker/impl/feel/FeelEngineWrapperExpressionEvaluationTest.java
@@ -50,6 +50,21 @@ class FeelEngineWrapperExpressionEvaluationTest {
   }
 
   @Test
+  void evaluateToJson_ShouldSucceed_WhenHandlingPojo() {
+    // given
+    final var resultExpression = "= { value: response.value, response: response }";
+    final var variables = new TestPojo("FOO");
+
+    // when
+    final String evaluatedResultAsJson =
+        objectUnderTest.evaluateToJson(resultExpression, variables);
+
+    // then
+    assertThat(evaluatedResultAsJson)
+        .isEqualTo("{\"response\":{\"value\":\"FOO\"},\"value\":\"FOO\"}");
+  }
+
+  @Test
   void evaluateToJson_ShouldSucceed_WhenExpressionStartsWithEqualsSign() {
     // given
     // FEEL expression -> ={"processedOutput":response.callStatus}
@@ -146,5 +161,18 @@ class FeelEngineWrapperExpressionEvaluationTest {
 
     // then
     assertThat(e.getMessage()).contains(FeelEngineWrapper.ERROR_EXPRESSION_EVALUATION_FAILED);
+  }
+
+  class TestPojo {
+
+    private String value;
+
+    public TestPojo(String value) {
+      this.value = value;
+    }
+
+    public String getValue() {
+      return value;
+    }
   }
 }

